inc/passwords/namespace.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace HM\Platform\Security\Passwords;
+
+use function HM\Platform\get_config;
+use WP_Error;
+use ZxcvbnPhp\Zxcvbn;
+
+/**
+ * Bootstrap.
+ */
+function bootstrap() {
+	add_action( 'check_passwords', __NAMESPACE__ . '\\enforce_password_strength', 10, 3 );
+	add_action( 'admin_print_styles-user-edit.php', __NAMESPACE__ . '\\hide_weak_password_prompt' );
+	add_action( 'admin_print_styles-profile.php', __NAMESPACE__ . '\\hide_weak_password_prompt' );
+}
+
+/**
+ * Get minimum required strength for passwords.
+ *
+ * @return int
+ */
+function get_minimum_strength() {
+	$config = get_config()['modules']['security'];
+
+	/**
+	 * Filter the minimum required password strength.
+	 *
+	 * Filter the required strength for a password. These scores correspond to
+	 * the following UI strings:
+	 *
+	 * - 1 = "Very Weak"
+	 * - 2 = "Weak"
+	 * - 3 = "Medium"
+	 * - 4 = "Strong"
+	 *
+	 * @param int $minimum_strength Minimum required strength. Defaults is 3.
+	 */
+	return apply_filters( 'hm-platform.security.passwords.minimum_strength', $config['minimum-password-strength'] );
+}
+
+/**
+ * Enforce minimum password strength during password check.
+ *
+ * @param string $user_login User login slug
+ * @param string $pass1 First password input field
+ * @param string $pass2 Second password input field
+ */
+function enforce_password_strength( $user_login, &$pass1, &$pass2 ) {
+	if ( empty( $pass1 ) || $pass1 !== $pass2 ) {
+		// Handled by WordPress, skip.
+		return;
+	}
+
+	// Add extra data to password strength checks.
+	// (Matches list in password-strength-meter.js)
+	$user = get_user_by( 'login', $user_login );
+	$extra_data = [
+		$user->user_login,
+		$user->first_name,
+		$user->last_name,
+		$user->nickname,
+		$user->display_name,
+		$user->email,
+		$user->url,
+		$user->description,
+	];
+
+	$checker = new Zxcvbn();
+	$results = $checker->passwordStrength( $pass1, $extra_data );
```

Comment:
Y U no HM Linter? Because this surely wouldn't pass it.

Reply:
This should definitely pass HM Linter; if not, the rule needs to be changed.

Reply:
camelCase is not allowed I thought.

Reply:
camelCase is fine when you're *using* a function/method, otherwise any time we interface with third-party code it'd fail. The checks are all done on the definitions.